tests/TestCase/ORM/TableTest.php

```diff
@@ -964,6 +964,137 @@ public function testDeleteAllFailure()
         $table->deleteAll(['id >' => 4]);
     }
 
+
+    /**
+     * @return array
+     */
+    public function dataDeleteEach()
+    {
+        return [
+            [['batch' => 1]],
+            [['batch' => 100]],
+            [['atomic' => false]],
+            [['stopOnFailure' => false]],
+            [['checkRules' => false]]
+        ];
+    }
+
+    /**
+     * @dataProvider dataDeleteEach
+     * @param array $options
+     */
+    public function testDeleteEach(array $options)
+    {
+        $table = new Table(['table' => 'users', 'connection' => $this->connection,]);
+
+        $beforeCount = 0;
+        $table->eventManager()->on('Model.beforeDeleteEach', function () use(&$beforeCount) {
+            $beforeCount++;
```

Comment:
You could increment an object property, that would save passing values by reference.